flight/PiOS/Common/pios_ir_transponder.c

```diff
@@ -0,0 +1,144 @@
+/**
+ ******************************************************************************
+ * @file       pios_ir_transponder.c
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2016
+ * @brief      Generate packets for various infrared lap timin protocols
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include <pios.h>
+
+#if defined(PIOS_INCLUDE_IR_TRANSPONDER)
+#include <misc_math.h>
+
+
+/**
+ * @brief Invert bits, needed for some protocols
+ */
+static void ir_invert_bits(uint8_t * data, uint8_t data_len)
+{
+	for (int i=0; i<data_len; i++) {
+		data[i] = ~data[i];
+	}
+}
+
+
+/* Based on code generated by pycrc v0.9, https://pycrc.org
```

Comment:
Ah yah, it's not.  Still-- can you move this into that file?  (which -- the other instances in there need a style cleanup / possibly to be changed to use nibble tables instead of these giant byte-oriented tables).